internal/translation/integrations/conversion.go

```diff
@@ -0,0 +1,106 @@
+package integrations
+
+import (
+	"context"
+	"fmt"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/cmp"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/common"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/project"
+)
+
+type Integration struct {
+	project.Integration
+}
+
+func (i Integration) Identifier() interface{} {
+	return i.Type
+}
+
+func NewIntegration(spec *project.Integration) (*Integration, error) {
+	if spec == nil {
+		return nil, nil
+	}
+	if err := cmp.Normalize(spec); err != nil {
+		return nil, fmt.Errorf("failed to normalize integration: %w", err)
+	}
+	return &Integration{Integration: *spec}, nil
+}
+
+func NewIntegrations(specs []project.Integration) ([]Integration, error) {
```

Comment:
I would remove this from here. Once we add the passwords as parameters in, this loop is probably better run with the reconcile code, getting all passwords and then passing them in.

Reply:
Wait, I think this is only used for comparisons, so it might be fine without filling passwords.

Maybe we could use 2 separate types with and without passwords instead.